### ISSUE TITLE
The toast doesn't disappear if progress is set to false

### ISSUE THREAD

**Issue by migliori** [xrr2016/tata#6]
Hi,

Thanks for this nice work.
Small bug (on your [demo](https://xrr2016.github.io/tata/) too):
The toast doesn't disappear if progress is set to false

**Comment by xrr2016** [comment]
> Hi,
> 
> Thanks for this nice work.
> Small bug (on your [demo](https://xrr2016.github.io/tata/) too):
> The toast doesn't disappear if progress is set to false

yes there is a bug, however it's to long time that i don't know how to fix it... 😓

**Comment by Sodium-Aluminate** [comment]
> > Hi,
> > Thanks for this nice work.
> > Small bug (on your [demo](https://xrr2016.github.io/tata/) too):
> > The toast doesn't disappear if progress is set to false
> 
> yes there is a bug, however it's to long time that i don't know how to fix it...

把隐藏进度条改为零px透明进度条暴力修复一下（雾

**Comment by lihas** [comment]
Any workarounds?

**Comment by lihas** [comment]
canging 
`function(t){return t===t}`

to 

`function(t){return t.id===m.id;}`

in https://github.com/xrr2016/tata/blob/master/dist/tata.js seems to fix the issue.
The issue is that when the library removes old toasts instead of checking for toast with an id it removes the very first toast from an array. This may lead to some toasts having their functionality broken if data for them was deleted

**Comment by lihas** [comment]
Created a pull request which should fix this issue https://github.com/xrr2016/tata/pull/8

**Comment by lihas** [comment]
`https://github.com/xrr2016/tata/pull/8`
